ykrt/src/location.rs

```diff
@@ -457,8 +460,16 @@ pub(super) struct ThreadIdInner;
 /// A `Location`'s non-counting states.
 #[derive(EnumDiscriminants)]
 pub(super) enum HotLocation {
+    /// Points to executable machine code that can be executed instead of the interpreter for this
+    /// HotLocation.
     Compiled(Box<CompiledTrace>),
-    Compiling(Arc<Mutex<Option<Box<CompiledTrace>>>>),
+    /// This HotLocation is being compiled in another thread.
+    Compiling,
+    /// Whilst this HotLocation was being compiled in another thread, the [Location] was dropped.
```

Comment:
why `[Location]`?

Reply:
That turns it into a link to the docs for the `Location` type.

Reply:
Ah. I did not know that. So should `HotLocation` (and all other type names throughout the repo) also have the square brackets?

Reply:
If you generate `cargo doc`s from this, that'll turn into a clickable link to `Location`.

Reply:
Yes, but why is `Location` special? Do the other types not get to have a clickable link too?

Reply:
There's not much advantage to having a reference to `Self` be clickable.